### PR TITLE
Fix leaking resources issue during build

### PR DESCRIPTION
**Pull request by adilliadil**
This change removes the resources that the built was not able to remove by itself.
These are the leaking resources currently:
1. Packer resource group
2. Temporary storage account
3. Managed Image for SIG scenarios

- [x] Tested for all the scenarios above

### CODE REVIEW

File: vhdbuilder/packer/cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+for v in "${required_env_vars[@]}"; do
+  if [ -z "${!v}" ]; then
+    echo "$v was not set!"
+    exit 1
+  fi
+done
+
+#clean up the temporary storage account
+id=$(az storage account show -n ${SA_NAME} -g ${AZURE_RESOURCE_GROUP_NAME} | jq .id)
```
Comment (yizhang4321) [inline vhdbuilder/packer/cleanup.sh:21]: Is SA_NAME a required env variable? Didn't see it in the list of "required_env_vars".

**Review by yizhang4321 — APPROVED** [review]

File: vhdbuilder/packer/cleanup.sh (hunk 1 above)
Comment (adilliadil) [reply vhdbuilder/packer/cleanup.sh:29]: Done!!

File: .pipelines/.vsts-vhd-builder.yaml [hunk 2]
```diff
@@ -122,11 +110,14 @@ phases:
...
+          -e PKR_RG_NAME=${PKR_RG_NAME} \
+          -e MODE=${MODE} \
           -e AZURE_RESOURCE_GROUP_NAME=${AZURE_RESOURCE_GROUP_NAME} \
-          ${CONTAINER_IMAGE} make -f packer.mk delete-mi
-        displayName: Clean-up Managed Image
-        condition: ne(variables['MODE'], 'default')
+          -e SA_NAME=${SA_NAME} \
+          -e IMAGE_NAME=${IMAGE_NAME} \
+          ${CONTAINER_IMAGE} make -f packer.mk cleanup
+        displayName: Clean Up Remaining Resources
```
Comment (xuto2) [inline .pipelines/.vsts-vhd-builder.yaml:119]: Packer/Temp resources

File: .pipelines/templates/.builder-release-template.yaml [hunk 3]
```diff
@@ -69,35 +69,26 @@ steps:
...
+      -e CLIENT_SECRET="$(CLIENT_SECRET)" \
+      -e TENANT_ID=${TENANT_ID} \
+      -e SUBSCRIPTION_ID="${SUBSCRIPTION_ID}" \
+      -e PKR_RG_NAME=${PKR_RG_NAME} \
+      -e MODE=${MODE} \
+      -e AZURE_RESOURCE_GROUP_NAME=${AZURE_RESOURCE_GROUP_NAME} \
+      -e SA_NAME=${SA_NAME} \
+      -e IMAGE_NAME=${IMAGE_NAME} \
+      ${CONTAINER_IMAGE} make -f packer.mk cleanup
+    displayName: Clean Up Remaining Resources
```
Comment (xuto2) [inline .pipelines/templates/.builder-release-template.yaml:90]: Packer/Temp

Comment (xuto2) [reply .pipelines/templates/.builder-release-template.yaml:90]: Also this is not enough, if the packer build failed in between, this clean up job will not run

**Pull request merged**